diff --git a/jni/private/BUILD.bazel b/jni/private/BUILD.bazel
index 5b85992..a015196 100644
--- a/jni/private/BUILD.bazel
+++ b/jni/private/BUILD.bazel
@@ -27,6 +27,7 @@ copy_file(
     src = "@com_github_openjdk_jdk_jni_h//file",
     out = "jni.h",
     is_executable = False,
+    visibility = ["//visibility:public"],
 )
 
 bzl_library(
diff --git a/jni/private/unix/BUILD.bazel b/jni/private/unix/BUILD.bazel
index 049be4d..947be90 100644
--- a/jni/private/unix/BUILD.bazel
+++ b/jni/private/unix/BUILD.bazel
@@ -5,5 +5,6 @@ copy_file(
     src = "@com_github_openjdk_jdk_unix_jni_md_h//file",
     out = "jni_md.h",
     is_executable = False,
-    visibility = ["//jni/private:__pkg__"],
+    # visibility = ["//jni/private:__pkg__"],
+    visibility = ["//visibility:public"],
 )
